test/unit/org/apache/cassandra/utils/JMXServerUtilsTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.utils;
+
+import java.nio.file.Paths;
+import javax.management.remote.JMXConnectorServer;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class JMXServerUtilsTest
+{
+    private static JMXConnectorServer jmxServer;
+
+    @BeforeClass
+    public static void setup() throws Exception
+    {
+        final String jmxRemoteSslPropertiesFile =
+            Paths.get(ClassLoader.getSystemResource("utils/jmxreamotessl.properties").toURI()).toString();
+
+        System.setProperty("com.sun.management.jmxremote.ssl", "true");
+        System.setProperty("cassandra.jmx.remote.ssl.config.file", jmxRemoteSslPropertiesFile);
+        jmxServer = JMXServerUtils.createJMXServer(9999, false);
+    }
+
+
+    @Test
+    public void testSetJmxRemoteSslPropertiesFromFileNoFileExists()
+    {
+        System.setProperty("cassandra.jmx.remote.ssl.config.file", "/bad/file/path");
+        try
+        {
+            jmxServer = JMXServerUtils.createJMXServer(9999, false);
+            fail("Test failed; should have thrown an exception");
+        }
+        catch (Exception e)
+        {
+            assertTrue(e.getClass().toString().contains("FileNotFoundException"));
+            assertTrue(e.getMessage().contains("JMX remote SSL configuration file does not exist: /bad/file/path"));
```

Comment:
You get much better failure messages if you use hamcrest for this:
```suggestion
            assertThat(e.getMessage(), containsString("JMX remote SSL configuration file does not exist: /bad/file/path"));
```